include/rmm/detail/stream.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstddef>
+
+// forward decl
+using cudaStream_t = struct CUstream_st*;
+
+namespace rmm {
+
+/**
+ * @brief Strongly-typed wrapper for CUDA streams with default constructor.
+ */
+class stream_t {
```

Comment:
I was hesitant to use `stream` as it could lead to awkward naming gymnastics where variables would otherwise also be named `stream`. I suppose this would only be an issue inside rmm, as elsewhere the type will have the namespace in front of it.
Would `stream` be ok, or is there a better suggestion?

Reply:
I was trying to think of another name. Some things that came to mind:
- `rmm::stream_view`
- `rmm::cuda_stream`

Reply:
I like `stream_view` and `managed_stream`

Reply:
What about `rmm::cuda_stream_view` for the view and `rmm::cuda_stream` for the owning type?

Reply:
Aim for convergent naming conventions between highly aligned types? :pray: